arkouda/decorators.py

```diff
@@ -0,0 +1,18 @@
+from typing import Union
+from arkouda import pdarray
+from arkouda.strings import Strings
+
+def checkforpdarray(func):
+    def check(pda : pdarray, **args):
+        if not isinstance(pda, pdarray):
+            raise TypeError('must be a pdarray, not a {}'.\
+                        format(pda.__class__.__name__))
+        return func(pda)
```

Comment:
I'm new to decorators, but do `args` need to be passed to `func` somehow?

Reply:
Good point, yes, and it's kinda handled under the covers so to speak

Reply:
@reuster986 I moved this code out and put the type checking decorators in [PR 543](https://github.com/mhmerrill/arkouda/pull/543)